### PR TITLE
patches: Convert incorrect patches

### PR DESCRIPTION
**Pull request by oliv3r**
OpenWRT's developer guide prefers having actual patches so they an be sent upstream more easily.

However, in this case, Adding proper fields also allows for `git am` to properly function. Some of these patches are quite old, and lack much traceable history.

This commit tries to rectify that, by digging in the history to find where and how it was first added.

It is by no means perfect and also shows some patches that should have been long gone.

Noted, that the histogram was done manually, so probably isn't 100% accurate, but it's only informal anyway.
Secondly, where git was very adamant about whitespace issues, these where corrected as git was refusing to accept them otherwise.

### CODE REVIEW

**Review by BKPepe — APPROVED** [review]
I like it. Anything what improves those hacky patches is useful. :+1:  Thanks!

**Comment by Ansuel** [comment]
@BKPepe one of the problem is the arbitrary sob... there was a similar discussion about hack patch and there were even some proposed solution about the authorship and sob

**Comment by Ansuel** [comment]
This is the issue https://github.com/openwrt/openwrt/issues/10272
and the pr https://github.com/openwrt/openwrt/pull/10264

**Comment by BKPepe** [comment]
Yes, I know, but I doubt anyone will do anything about those hack patches, which have been here for some time and I am sure that if you do SoB for the commit message, then it might be applied for the patch itself when the commit introduced it. But SoB in this PR is just added for some patches, so if you insist on it, then it can be removed and merged when there is added commit subject and description.

This at least adds an easy way to reach the author of the patch to reach him while manual rebasing is problematic.

**Comment by Ansuel** [comment]
@BKPepe i guess we can permit sob as any change to that patch was signed by the commit that introduced the change itself but I still think we should ask for permission to add the tag. (And this will be very slow)

Wonder if we should add a dummy sob and be done with it? (if we don't get any approval on the tag?)

I'm not against this change, concern is just adding sob tag without real permission and that can be problematic if someone doesn't want to take responsibility for a patch.

**Comment by BKPepe** [comment]
What you are proposing can take a long time, but what we agreed on that those patches should be revisited/improved and done something with them. So, I leave it up to you what's you like, so we don't argue about that, but I am not a lawyer. That's why I think someone else should decide on it and pull the trigger. Maybe, it could be discussed on OpenWrt mailing list?

> I'm not against this change, concern is just adding sob tag without real permission and that can be problematic if someone doesn't want to take responsibility for a patch.

Ehh, I maybe would have a different opinion on this one, which you would not like, but we should remember why the patch was introduced and if we really need it. If he does not take responsibility for the patch, it can be thrown away. :-) Simple as that. Otherwise, we would not improve the current situation. I know it will take time, and it can be a can of worms.

**Comment by Ansuel** [comment]
> Maybe, it could be discussed on OpenWrt mailing list?

Yep seems a nice approach. I will push a mail on the mailing list about that referencing this pr hoping we get some feedback. Should bet more attention than github for sure.

I would way some days and then if nobody is against I would just add the sob tag following the rule "you make the change you also take responsibility for the patch"

**Comment by oliv3r** [comment]
> @BKPepe one of the problem is the arbitrary sob... there was a similar discussion about hack patch and there were even some proposed solution about the authorship and sob

I did my best to 'take' the SoB of the original commit; I spend a good 2 days on making it 'as authentic as possible'. So the question then remains, what we had before, vs what there is now, this surely must be an improvement.

**Comment by oliv3r** [comment]
> Soo some patch can even be dropped...

I noticed this as well so this could be an opportunity to clean some of these up!

**Comment by oliv3r** [comment]
> I'm not against this change, concern is just adding sob tag without real permission and that can be problematic if someone doesn't want to take responsibility for a patch.

Interesting point, and I certainly spent time/thoughts on this as well. The way I see it; the patch was given to OpenWRT in the past, with an SoB; copying that SoB from the original commit should be fine.

Patches that lacked an SoB, patches just dropped in the tree, but where the dropping commit had an SoB, I copied that, and also, should be ok.

Since I'd done the work a while ago, I don't recall what I did with commits that lacked SoB, and had zero traceability. Those basically will be without SoB. I think it's fine, because they where without before, so just having some minor improvement in patch format would already be a win?

Also, it's not like we'll upstream these fixed patches ad-verbatim.

**Comment by oliv3r** [comment]
I think I miss something though; as I recall I had a handful of 5.15 patches as well, so I'll have to look through my stash for those :p

**Comment by oliv3r** [comment]
Found them :) Also rebased new patches :( apparently we still add hacky patches :(

**Comment by oliv3r** [comment]
Btw, I don't mind splitting my patches into 3 once a decision has been made (1 generic, 1 ath79, 1 realtek) and we could even do 3 MR's if we prefer that for cleanliness ...

**Comment by Ansuel** [comment]
@oliv3r yes if it's not too much pls split the commits. Also with a quick feedback in IRC and mailing list we agreed that SoB is too much but we can totally use other tag that will still reference to the creator without adding a SoB.

From jow

> Hi,
> 
> to be pragamtic, set a `Submitted-by`, `Added-by`, `Introduced-by` or similar
> tag. This should provide enough clue to locate the original author without the
> need to "forge" Sob.

**Comment by oliv3r** [comment]
Not sure how I can fix the 'formalities' here
```
/home/runner/work/_temp/8b824caf-e30c-4dec-9add-c7488bed36e3.sh: line 1: .github/workflows/scripts/ci_helpers.sh: No such file or directory
```
sounds like a CI issue, re-running it via force push might fix this?

**Comment by Ansuel** [comment]
@oliv3r we changed the location of that script but your repo still is not updated sooo you just have to rebase this on top of master

**Comment by oliv3r** [comment]
> @oliv3r yes if it's not too much pls split the commits. Also with a quick feedback in IRC and mailing list we agreed that SoB is too much but we can totally use other tag that will still reference to the creator without adding a SoB.
> 
> From jow
> 
> > Hi,
> > to be pragamtic, set a `Submitted-by`, `Added-by`, `Introduced-by` or similar
> > tag. This should provide enough clue to locate the original author without the
> > need to "forge" Sob.

I'm okay going through the (considerable) effort on re-writing the SoB; but most, if not all, came from 'git format-patch' so we'd be 'changing' that. Is that okay? If so, I can do a 'sed' o the patchset, and replace Signed-off-by -> Submitted-by signoff, but it would have to be 'that pragmatic' indeed. Would that be okay @ansuel?

**Comment by Ansuel** [comment]
@oliv3r IMHO we should just change any sob with submitted-by. A quick way would be split the commit first... then just create the patch, make the change and reapply them with git am... Quick and dirty ahahaha

We keep track of who pushed the pr without adding a sob that can cause problems. imho it's the best solution... (I mean the other one is to NOT include any tag at all... sooo)

**Comment by oliv3r** [comment]
> @oliv3r IMHO we should just change any sob with submitted-by. A quick way would be split the commit first... then just create the patch, make the change and reapply them with git am... Quick and dirty ahahaha
> 
> We keep track of who pushed the pr without adding a sob that can cause problems. imho it's the best solution... (I mean the other one is to NOT include any tag at all... sooo)

Just as I wanted to do it then; though I'd do the sed first, split second, so I have 1 patch to 'sed' on :)

**Comment by oliv3r** [comment]
@Ansuel they are split-up now, and rebased. Didn't check if more hacky patches where added in the meantime :(

**Comment by Ansuel** [comment]
@oliv3r do you have time to check if more patch have been applied or no time and we merge this now?

**Comment by oliv3r** [comment]
@Ansuel don't forget to come out of the shower ;)

**Comment by Ansuel** [comment]
@oliv3r my god you will hate me... more patch has changed and also for the generic commit you should use generic: instead of patches: 

:(

**Comment by oliv3r** [comment]
> @oliv3r my god you will hate me... more patch has changed and also for the generic commit you should use generic: instead of patches:
> 
> :(

ah, yeah I think saw that, but forgot; i changed it for the other targets :p Let me work some magic

**Comment by oliv3r** [comment]
There, rebased and updated :)

**Pull request merged**
